src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java

```diff
@@ -57,7 +55,49 @@ public void testConnector_connectorFailsWithNoTopicsGiven() {
         final String channelName = "test-channel";
 
         Map<String, String> settings = createSettings(channelName, "some_fake_key", "some_client_id", null);
-        assertThrows(org.apache.kafka.connect.runtime.rest.errors.ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
+        assertThrows(ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
+    }
+
+    @Test
+    public void testConnector_connectorFailsWithNoClientIdGiven() {
+        final String channelName = "test-channel";
+
+        Map<String, String> settings = createSettings(channelName, "some_fake_key", null, TOPICS);
+        assertThrows(ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
+    }
```

Comment:
I'd suggest refactoring the new tests with the "GIVEN WHEN THEN" approach and make sure that the important things for a given test are highlighted :wink: For example, here you're declaring a `channelName` at the start of the test, so I'm already thinking "okay, we'll be testing how this channel name affects the behaviour" but in this test the channel name is not important :sweat_smile: After the suggested changes this test could look like this for example:
```java
    @Test
    public void testConnector_connectorFailsWithNoClientIdGiven() {
        // given
        final String clientId = null;

        // when
        Map<String, String> settings = createSettings("test-channel", "some_fake_key", clientId, TOPICS);

        // then
        assertThrows(ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
    }
```

Reply:
@KacperKluka A bit busy right now, but I shall put those onto organised states later in another PR